docs/howto/craft-a-snap/example-cross-compiled-app.rst

```diff
@@ -0,0 +1,161 @@
+.. _example-cross-compiled-app:
+
+Example Cross-Compiled App
+==========================
+
+Projects built using the :ref:`craft_parts_autotools_plugin` can be cross-compiled,
+allowing software to be developed on one architecture and built for a number of target
+architectures.
+
+Overview
+--------
+
+There are a number of steps to enable cross-compilation for an autotools project:
+
+#. Specify the architectures used.
+#. Adjust the options passed to the ``configure`` script.
+#. Include packages to support cross-compilation.
+#. Handle linker warnings.
+
+Example project file for example-curl
+-------------------------------------
+
+The following code comprises the project file for a project that uses ``libcurl``. The
+complete code can be found in `this repository <https://github.com/mr-cal/hello-curl>`_.
+
+.. collapse:: hello-curl project
+
+  .. literalinclude:: ../code/craft-a-snap/example-cross-compiled-app-recipe.yaml
+      :caption: snapcraft.yaml
+      :language: yaml
+
+The ``platforms`` key
+-------------------------
+
+The :ref:`platforms <architectures-howto>` key is normally used to specify which
+architectures should be used when building a snap. For example, in the following case
+the build architecture is the same as the target architecture:
+
+.. code:: yaml
+
+    platforms:
+      - amd64
+
+It is also possible to specify different architectures for the build process and the
+target system. This enables cross-compilation to be managed by Snapcraft.
+
+Enable cross-compilation
+------------------------
+
+The ``platforms`` key can also be used to specify sets of architectures using the
+``build-on`` and ``build-for`` sets. In the simplest case, this can be used to
+cross-compile on one architecture for a single, different architecture:
+
+.. code:: yaml
+
+    platforms:
+      - build-on: amd64
+        build-for: armhf
+
+In this example, Snapcraft will only build the project on the ``amd64`` architecture,
+and it will only build it for the ``armhf`` architecture.
```

Comment:
In line with my earlier comment about curl, `amd64` and `armhf` are machine names for the semi-official terms "AMD64" and "armhf".

A corollary would be the following fabricated sentences:

> To build for the `win` and `darwin` platforms, list `win` and `darwin` in the `platforms` key.

Compared to:

> To build for Windows and macOS, list `win` and `darwin` in the `platforms` key.